interceptor/interceptor.go

```diff
@@ -0,0 +1,221 @@
+// The MIT License
+//
+// Copyright (c) 2021 Temporal Technologies Inc.  All rights reserved.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+// Package interceptor contains interceptors for client and worker calls.
+package interceptor
+
+import (
+	"context"
+
+	commonpb "go.temporal.io/api/common/v1"
+	"go.temporal.io/sdk/internal"
+	"go.temporal.io/sdk/workflow"
+)
+
+// Interceptor is a common interface for all interceptors. It combines
+// ClientInterceptor and WorkerInterceptor. If an implementation of this
+// interceptor is provided via client options, some client calls and all worker
+// calls will be intercepted by it. If an implementation of this interceptor is
+// provided via worker options, all worker calls will be intercepted by it.
+//
+// All implementations of this should embed InterceptorBase but are not required
+// to.
+type Interceptor = internal.Interceptor
+
+// InterceptorBase is a default implementation of Interceptor meant for
+// embedding. It simply embeds ClientInterceptorBase and WorkerInterceptorBase.
+type InterceptorBase = internal.InterceptorBase
```

Comment:
Go lint is complaining about this stuttering and breaking the build. I am going to change the linter, not this (it is a stubborn linter, so I may not be able to easily).

Reply:
You can just disable individual lines with a comment I believe, too

Reply:
Not with golint. It is deprecated and the authors stubbornly and _very_ intentionally chose not to allow such a thing. The vast majority of projects have moved off of it (or at least don't error on it).  See https://github.com/golang/lint/issues/263, https://github.com/golang/go/issues/38968, etc.

Reply:
Golint removed from `make check`